install/update.php

```diff
@@ -2607,15 +2849,19 @@ function do_biosascomponentmigration() {
          $key = md5($data['bios_date'] . $data['bios_version']. $data['bios_manufacturers_id']);
          if (!isset($bioses[$key])) {
             //look for an existing BIOS in the database
-            $query = "SELECT id FROM glpi_devicefirmwares
-                        WHERE
-                           date = '{$data['bios_date']}'
-                           AND version = '{$data['bios_version']}'
-                           AND manufacturers_id = '{$data['bios_manufacturers_id']}'
-                           LIMIT 1 OFFSET 0";
-            $result = $DB->query($query);
-            if ($DB->numrows($result)) {
-               $existing = $DB->fetch_assoc($result);
+            $iterator = $DB->request([
+               'SELECT' => 'id',
+               'FROM'   => 'glpi_devicefirmwares',
+               'WHERE'  => [
+                  'date'               => $data['bios_date'],
+                  'version'            => $data['bios_version'],
+                  'manufacturers_id'   => $data['bios_manufacturers_id']
+               ],
+               'START'  => 0,
```

Comment:
START/OFFSET differs from LIMIT/OFFSET.
And OFFSET exists as crit in iterator ?

Reply:
OFFSET does not exists; I should write LIMIT...